tools/generator/src/Extensions/XcodeScheme+Extensions.swift

```diff
@@ -0,0 +1,167 @@
+// MARK: TargetWithID
+
+extension XcodeScheme {
+    struct TargetWithID {
+        let id: TargetID
+        let target: Target
+    }
+}
+
+extension XcodeScheme.TargetWithID: Equatable {
+    static func == (lhs: XcodeScheme.TargetWithID, rhs: XcodeScheme.TargetWithID) -> Bool {
+        return lhs.id == rhs.id
+    }
+}
+
+extension XcodeScheme.TargetWithID: Comparable {
+    static func < (lhs: XcodeScheme.TargetWithID, rhs: XcodeScheme.TargetWithID) -> Bool {
+        return lhs.target.platform < rhs.target.platform
+    }
+}
+
+// MARK: Resolve TargetIDs
+
+extension XcodeScheme {
+    /// Represents a configuration string (Target.configuration).
+    typealias Configuration = String
+
+    /// Determines the mapping of `LabelValue` to the `TargetID` values based upon the scheme's
+    /// configuration.
+    func resolveTargetIDs(targets: [TargetID: Target]) throws -> [LabelValue: TargetID] {
```

Comment:
@brentleyjones The primary resolution logic is going to return a single `TargetID` per `LabelValue`. In the case when someone defines a "kitchen sink" scheme (i.e., put all targets for the `multiplatform` example in a single scheme), which of the platforms would you expect to win? Based upon the sort logic in `Platform`, it will be iOS simulator. What would you expect to happen in Xcode?

Reply:
If many different platform tests are listed as top level targets, we could find each platform version of a listed library, but just finding the "best" one per Platform sorting logic would be an acceptable first version.

Reply:
The reason that I ask is that I originally had this API return a `[LabelValue: [TargetID]]`. I can go back to that version if that is what we want.